packages/abi-typegen/src/abi/types/EnumType.ts

```diff
@@ -40,6 +42,19 @@ export class EnumType extends AType implements IType {
     return name;
   }
 
+  public getNativeEnum(typeHash: RawTypeHash) {
+    const { components } = this.rawAbiType;
+
+    // `components` array guaranteed to always exist for structs/enums
+    const enumComponents = components as IRawAbiTypeComponent[];
+
+    if (!enumComponents.every(({ type }) => !typeHash[type])) {
+      return undefined;
+    }
+
+    return enumComponents.map(({ name }) => `${name}`).join(', ');
+  }
+
```

Comment:
```suggestion
  public getNativeEnum(params: { types: IType[] }) {
    const { types } = params;

    const { components } = this.rawAbiType;

    // `components` array guaranteed to always exist for structs/enums
    const enumComponents = components as IRawAbiTypeComponent[];

    if (!enumComponents.every(({ type: typeId }) => !findType({ types, typeId })) {
      return undefined;
    }

    return enumComponents.map(({ name }) => `${name}`).join(', ');
  }

```

Reply:
Also, please add two test cases on the `EnumType.ts` file, one covering each `return` scenario.